apps/emqx_bridge_dynamo/src/emqx_bridge_dynamo_connector.erl

```diff
@@ -178,14 +178,34 @@ on_batch_query(InstanceId, [{_ChannelId, _} | _] = Query, State) ->
 on_batch_query(_InstanceId, Query, _State) ->
     {error, {unrecoverable_error, {invalid_request, Query}}}.
 
-on_get_status(_InstanceId, #{pool_name := Pool}) ->
+health_check_timeout() ->
+    15000.
+
+on_get_status(_InstanceId, #{pool_name := Pool} = State) ->
     Health = emqx_resource_pool:health_check_workers(
-        Pool, {emqx_bridge_dynamo_connector_client, is_connected, []}
+        Pool,
+        {emqx_bridge_dynamo_connector_client, is_connected, [
+            emqx_resource_pool:health_check_timeout()
+        ]},
+        health_check_timeout(),
+        #{return_values => true}
     ),
-    status_result(Health).
+    case Health of
+        {error, timeout} ->
+            {?status_connecting, State, <<"timeout_while_checking_connection">>};
+        {ok, [_ | _] = Results} ->
```

Comment:
This will have a clause match error if the list is empty when there are no workers.

Reply:
Can this happen?

Reply:
Added commit with a fix: https://github.com/emqx/emqx/pull/12678/commits/c90f4f579410135fb3ee01acbf268f696be8906f